### ISSUE TITLE
Multiple clients and user streams not working, only a single user stream works

### ISSUE THREAD

**Issue by ProjectDent** [tweetstream/tweetstream#194]
I've setup a simple test app which I'm running on the command line:

```
require 'tweetstream'

first_credentials = {
	consumer_key: 'ABC',
	consumer_secret: 'DEF',
	oauth_token: 'GHI',
	oauth_token_secret: 'JKL'
}

first_client = TweetStream::Client.new(first_credentials)

first_client.on_direct_message do |direct_message|
	puts 'first client, direct message: ' + direct_message.text
end

first_client.userstream

second_credentials = {
	consumer_key: 'ABC',
	consumer_secret: 'DEF',
	oauth_token: 'MNO',
	oauth_token_secret: 'PQR'
}

second_client = TweetStream::Client.new(second_credentials)

second_client.on_direct_message do |direct_message|
	puts 'second client, direct message: ' + direct_message.text
end

second_client.userstream
```

I've removed the real keys and tokens. When I run this code, I get direct messages that come through to the first client, but not the second client. If I switch the order, I get the direct messages that come through to the second client, but not the first.

So it seems that after starting the userstream for a client, a userstream can't be added for additional clients.

I don't have access to the Site Streams API, and Twitter likely won't be allowing me access, so that's not an issue.

Since these clients are separate objects, I wouldn't have expected them to affect each other. Is there something I'm doing wrong, or is this intentional? Having multiple user streams at a time is crucial to my app, so I'm hoping there's a way to make this work.

**Comment by stve** [comment]
hi @ProjectDent 

you might want to try something like this:

```ruby

require 'tweetstream'

first_credentials = {
    consumer_key: 'ABC',
    consumer_secret: 'DEF',
    oauth_token: 'GHI',
    oauth_token_secret: 'JKL'
}

first_client = TweetStream::Client.new(first_credentials)

first_client.on_direct_message do |direct_message|
    puts 'first client, direct message: ' + direct_message.text
end

second_credentials = {
    consumer_key: 'ABC',
    consumer_secret: 'DEF',
    oauth_token: 'MNO',
    oauth_token_secret: 'PQR'
}

second_client = TweetStream::Client.new(second_credentials)

second_client.on_direct_message do |direct_message|
    puts 'second client, direct message: ' + direct_message.text
end

EM.run do 
  first_client.userstream
  second_client.userstream
end
```

If that doesn't work, you'll need to run each in a separate thread in order to have concurrent streams.

**Comment by ProjectDent** [comment]
Including the `userstream` calls in the `EM.run` block _does_ work, except that I'll want to add additional clients later on, without restarting the setup.

Having multiple EM.run blocks means it doesn't run the second one.

**Comment by ProjectDent** [comment]
Second suggestion on separate threads worked.

**Comment by ProjectDent** [comment]
I thought this was working - perhaps I was wrong.

I have the following code sample:

```
require 'tweetstream'

Thread.new do
	#AledSpaceJones
	first_credentials = {
		consumer_key: 'ABC',
		consumer_secret: 'DEF',
		oauth_token: '123',
		oauth_token_secret: 'XYZ'
	}

	first_client = TweetStream::Client.new(first_credentials)

	first_client.on_anything do |something|
		puts 'something'
	end

	puts 'first user stream'
 	first_client.userstream
end

Thread.new do
	#ProjDentPaul
	second_credentials = {
		consumer_key: 'ABC',
		consumer_secret: 'DEF',
		oauth_token: '456',
		oauth_token_secret: 'XYZ'
	}

	second_client = TweetStream::Client.new(second_credentials)

	second_client.on_anything do |something|
		puts 'something else'
	end

	puts 'second user stream'
	second_client.userstream
end

thread = Thread.new do
	sleep 1000
end

thread.join
```

When I run this in console, I get:

first user stream
second user stream
something

So both threads are being entered, but the second user stream doesn't appear to be setup, as the `on_anything` call is never made, when it's usually made immediately upon setup. Additionally, sending a direct message or tweet to that account doesn't print anything.

Is there any way I can diagnose why the second user stream isn't being setup? Anything I'm doing incorrectly?

My purpose here is simply to have 2 user streams running at the same time.

**Comment by ProjectDent** [comment]
Actually, I again commented out the first half of the code, and the second half doesn't work. Which means it's a problem with the second set of credentials. But I checked using my iOS component and they're legitimate as far as I can tell. I added a bunch of error detection blocks (on_unauthorized, on_error, on_no_data_received etc.) and also on_init. Well, on_init gets called. And then after that, on_something doesn't get called, and no does anything else.
